### ISSUE TITLE
Rework debug logging

### ISSUE THREAD

**Issue by lynxlynxlynx** [ja2-stracciatella/ja2-stracciatella#60]
Originally reported by: **Gennady Trafimenkov (Bitbucket: [gennady](http://bitbucket.org/gennady), GitHub: [gennady](http://github.com/gennady))**

----------------------------------------

There is a lot of code like this littered all over the place:

```
			#ifdef JA2BETAVERSION
				ScreenMsg( FONT_MCOLOR_LTYELLOW, MSG_ERROR, L"Turning off AI...
```

```
				if (gfTurnBasedAI)
				{
					DebugAI( String("Ending turn for %d ...
```

```
					DebugMsg( TOPIC_JA2, DBG_LEVEL_3, String("Attack busy count ...
```

It would be great to develop a unifed scheme of dealing with such debug messages.

Requirements:
- logging to console
- logging to a file
- ? logging to the screen
- ability to turn logging off and on at run time, not compile time

----------------------------------------
- Bitbucket: https://bitbucket.org/gennady/ja2-stracciatella/issue/60

**Comment by derSteFfi** [comment]
so, as for now, we use SLOG (which was implemented by gennady). We discussed this topic in the meeting too. We agreed we have 2 partly seperated issues: rework debugging logs to use a unified scheme and decide which scheme it should be.

For now, we agreed SLOG offers everything we want to have. So whereever we touch files/functions, we should switch the logging to SLOG. I will start to convert the debug messages behind JA2[TEST,BETA]VERSION to SLOG calls with an appropriate level. This way we also solve part of #10 without actually hurting anything.
